exchanges/binance/binance_websocket.go

```diff
@@ -240,6 +240,10 @@ func (b *Binance) wsHandleData(respRaw []byte) error {
 				if err != nil {
 					return err
 				}
+				clientOrderID := data.Data.ClientOrderID
+				if oStatus == order.Cancelled {
+					clientOrderID = data.Data.CancelledClientOrderID
```

Comment:
From a user perspective and especially when interacting via GRPC, it would be nice to keep the ClientOrderID teh same even when the order is cancelled to allow for clear lookup. With Binance being different to most of the other exchanges, we could maybe think about a "NewCancellationId" or something?
At least for the typical user, they can go through the active and cancelled orders the same way then and don't have to be made aware that Binance is different.

Reply:
I can see the simplicity in overwriting the clientOrderID to match whatever the latest is in the short term however I like the idea of an additional "NewCancellationID / CancelledClientOrderID" field for exchanges like Binance. Preserving all information helps with auditing and debugging. Can leave this as-is though and have it addressed in a separate PR.